lib/modules/manager/cargo/locked-version.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { DirectoryResult, dir } from 'tmp-promise';
+import { join } from 'upath';
+import { Fixtures } from '../../../../test/fixtures';
+import { GlobalConfig } from '../../../config/global';
+import type { RepoGlobalConfig } from '../../../config/types';
+import { writeLocalFile } from '../../../util/fs';
+import { extractLockFileVersions, parseLockFile } from './locked-version';
+
+describe('modules/manager/cargo/locked-version', () => {
+  describe('extractLockFileVersions()', () => {
+    let adminConfig: RepoGlobalConfig;
+    let tmpDir: DirectoryResult;
+
+    beforeEach(async () => {
+      tmpDir = await dir({ unsafeCleanup: true });
+      adminConfig = {
+        localDir: join(tmpDir.path, 'local'),
+      };
+
+      GlobalConfig.set(adminConfig);
+    });
+
+    afterEach(async () => {
+      await tmpDir.cleanup();
+      GlobalConfig.reset();
+    });
```

Comment:
just mock `util/fs` instead. it's much simpler and faster than copying files around

Reply:
@viceice A newbie question: could you please help me out with the mock?

I have used `jest.mock('../../../util/fs');` at the top of the file, and it works for all `util/fs` calls within this module, but not for transitive calls that also indirectly uses it.

For example, in

```ts
    it('extracts registry urls from .cargo/config.toml', async () => {
      await writeLocalFile('.cargo/config.toml', cargo6configtoml);
      const res = await extractPackageFile(cargo6toml, 'Cargo.toml', {
        ...config,
      });
      expect(res?.deps).toMatchSnapshot();
      expect(res?.deps).toHaveLength(3);
    });
```

The `extractPackageFile()` call (which depends on `readLocalFile()`) cannot correctly read the file written by `writeLocalFile()` in the previous line, which causes this test to fail.

Reply:
Oh, never mind, looks like I can use `fs.readLocalFile.mockResolvedValueOnce()` instead of `writeLocalFile()`.